src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,112 @@
-export const Pagination = () => {};
+import React from 'react';
+import cn from 'classnames';
+
+import { getNumbers } from '../../utils';
+
+interface Props {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const itemsPerPage = getNumbers(
+    perPage * currentPage - perPage + 1, currentPage * perPage,
+  );
+
+  const itemsLink = getNumbers(1, Math.ceil(total / perPage));
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={cn(
+            'page-item',
+            { disabled: currentPage === 1 },
+          )}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage === 1}
+            onClick={() => {
+              if (currentPage > 1) {
+                onPageChange(currentPage - 1);
+              }
+            }}
```

Comment:
It's better to move such handlers to the separate handlers:
```
const handlePageChangeClick = () => {
  if (currentPage > 1) {
    onPageChage(currentPage - 1);
  }
};

// and then just pass this function as a prop:

<a
  ...
  onClick={handlePageChangeClick}
/>
```

And your code will be cleaner and readable.

Reply:
And you can use such an approach to any handlers below.